packages/server/src/requestPipeline.ts

```diff
@@ -431,13 +436,25 @@ export async function processGraphQLRequest<TContext extends BaseContext>(
             extensions: e.extensions,
           });
         }
+
+        if (isUnknownOperationNameError(e)) {
+          return new UnknownOperationNameError(e.message, {
+            nodes: e.nodes,
+            originalError: e.originalError,
+            extensions: e.extensions,
+          });
+        }
         return e;
       });
 
       if (resultErrors) {
         await didEncounterErrors(resultErrors);
       }
 
+      if (resultErrors?.length && !('data' in result)) {
+        return await sendErrorResponse(resultErrors);
```

Comment:
this calls didEncounterErrors, so we don't want to do that twice.

also this doesn't call executionDidEnd? that seems wrong?